src/keywords/dynamicsitenodes.cpp

```diff
@@ -44,7 +45,7 @@ bool DynamicSiteNodesKeyword::read(LineParser &parser, int startArg, const CoreD
         return Messenger::error("Parent ProcedureNode not set, so can't read DynamicSiteNode data.\n");
 
     // Create a new DynamicSite and add it to our data RefList
-    auto *dynamicSite = new DynamicSiteProcedureNode(parentNode_);
+    auto dynamicSite = std::dynamic_pointer_cast<DynamicSiteProcedureNode>(parentNode_->shared_from_this());
```

Comment:
This fixes the error in the CI for the `calculate_dangle` test.
```suggestion
    auto dynamicSite = std::make_shared<DynamicSiteProcedureNode>(
        std::dynamic_pointer_cast<SelectProcedureNode>(parentNode_->shared_from_this()));
```